### PR TITLE
Add Redshift usage crawler [sc-3968]

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?
To add Redshift usage crawler.

### 🤓 What?

- Add `metaphor.reshift.usage.extractor.RedshiftUsageExtractor`
- Add testcases

### 🧪 Tested?

The local test successfully.

### CODE REVIEW

File: metaphor/redshift/usage/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Redshift Usage Statistics Connector
+
+This connector extract Redshift usage statistics from a Redshift database using [asyncpg](https://github.com/MagicStack/asyncpg) library.
```
Comment (mars-lan) [inline metaphor/redshift/usage/README.md:3]:
```suggestion
This connector extractS Redshift usage statistics from a Redshift database using [asyncpg](https://github.com/MagicStack/asyncpg) library.
```

File: metaphor/redshift/usage/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+# (Optional) A list of regular expressions to limit the scan to selected datasets(schemas)
+dataset_filters:
+    - my_dataset
+    - prefix_.*
+```
+
+## Testing
+
+Follow the [Installation](../../README.md) instructions to install `metaphor-connectors` in your environment (or virtualenv). Make sure to include either `all` or `bigquery` extra.
```
Comment (mars-lan) [inline metaphor/redshift/usage/README.md:30]:
```suggestion
Follow the [Installation](../../README.md) instructions to install `metaphor-connectors` in your environment (or virtualenv). Make sure to include either `all` or `redshift` extra.
```

File: metaphor/redshift/usage/config.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+@deserialize
+@dataclass
+class RedshiftUsageRunConfig(RedshiftRunConfig):
+    lookback_days: int = 30
+
+    # Query filter to exclude certain usernames from the processing
+    excluded_usernames: Set[str] = dataclass_field(default_factory=lambda: set())
+
+    # Filters for dataset names (any match will be included)
+    dataset_filters: List[str] = dataclass_field(default_factory=lambda: [r".*"])
```
Comment (mars-lan) [inline metaphor/redshift/usage/config.py:19]: This is okay for now, but bear in mind that ultimately we'd want to unify the dataset filters using the same mechanism: https://github.com/MetaphorData/connectors/tree/main/metaphor/snowflake#filter

Comment (elic-eon) [reply metaphor/redshift/usage/config.py:19]: Got it, BigQuery also needs to use the same mechanism.

File: tests/redshift/usage/test_extractor.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    # @patch doesn't work for async func in py3.7: https://bugs.python.org/issue36996
+    with patch(
+        "metaphor.redshift.usage.extractor.RedshiftUsageExtractor._fetch_usage"
+    ) as mock_fetch_usage:
+
+        def fake(config):
+            return AsyncIter(records)
+
+        mock_fetch_usage.side_effect = fake
```
Comment (mars-lan) [inline tests/redshift/usage/test_extractor.py:69]: Might be easier to just use a lambda expression here?

**Review by mars-lan — COMMENTED** [review]
Also remember to bump up the version in pyproject.toml.

File: tests/redshift/usage/test_extractor.py (hunk 4 above)
Comment (elic-eon) [reply tests/redshift/usage/test_extractor.py]: Yes, using a lambda is better.

**Review by mars-lan — APPROVED** [review]

**Pull request merged**